src/components/search/Search.js

```diff
@@ -1,14 +1,81 @@
-import { SearchAlt2 } from '@styled-icons/boxicons-regular'
+import '@reach/combobox/styles.css'
+
+import {
+  Combobox,
+  ComboboxInput,
+  ComboboxList,
+  ComboboxOption,
+} from '@reach/combobox'
+import { useHistory } from 'react-router-dom'
+import usePlacesAutocomplete, {
+  getGeocode,
+  getLatLng,
+} from 'use-places-autocomplete'
+import { NumberParam, useQueryParams } from 'use-query-params'
 
 import Input from '../ui/Input'
+import SearchEntry from './SearchEntry'
+
+const Search = () => {
+  const [centerCoords, setCenterCoords] = useQueryParams({
+    centerLat: NumberParam,
+    centerLng: NumberParam,
+  })
+
+  let history = useHistory()
+
+  const onSelectHandler = (item) => {
+    getLongitudeAndLatitudeFromAddress(item)
+    history.push({
+      pathname: '/map',
+    })
+  }
+
+  const getLongitudeAndLatitudeFromAddress = (placeId) => {
+    getGeocode({ placeId })
+      .then((results) => getLatLng(results[0]))
+      .then(({ lat, lng }) => {
+        setCenterCoords(
+          { ...centerCoords, centerLat: lat, centerLng: lng },
+          'push',
+        )
+      })
+  }
+
+  const {
+    ready,
+    value,
+    suggestions: { status, data },
+    setValue,
+  } = usePlacesAutocomplete()
+
+  const handleInput = (e) => {
+    setValue(e.target.value)
+  }
 
-const Search = () => (
-  <Input
-    placeholder="Search for a location..."
-    onChange={(e) => console.log(e.target.value)}
-    onEnter={(e) => window.alert(`Received:\n${e?.target?.value}`)}
-    icon={<SearchAlt2 />}
-  />
-)
+  return (
+    <Combobox onSelect={onSelectHandler} aria-labelledby="demo">
+      <ComboboxInput
+        as={Input}
+        value={value}
+        onChange={handleInput}
+        disabled={!ready}
+      />
+      <ComboboxList>
+        {status === 'OK' &&
+          data.map(
+            ({
+              place_id,
+              structured_formatting: { main_text, secondary_text },
+            }) => (
+              <ComboboxOption as={SearchEntry} key={place_id} value={place_id}>
+                {[main_text, secondary_text]}
```

Comment:
Consider using a template string here for clarity.

Reply:
@SirajChokshi this has to do with the specific way SearchEntry wraps ListEntry to make it compatible with ComboboxOption, which passes data in via the children prop. Definitely interested if you have any better ideas on how to do this.

Reply:
I do agree this is a bit weird haha.